src/lib/get-user-variable.ts

```diff
@@ -0,0 +1,21 @@
+import * as RMOAS from '../rmoas.types';
+
+export type UserVariable = unknown | null;
+
+function getKey(key: RMOAS.User | Record<string, primitive>, property: string): UserVariable {
```

Comment:
nit: generally I'm against return types as all it really does is create redundant code that at best does nothing but at worst widens the type definition specificity

Reply:
obviously there are exceptions to this rule like for callback functions or something where the function ends up being defined elsewhere and it then needs to be consumed in some place and that consumer needs to guard against that return type

Reply:
Aren't return types the strongest part of TS?

Reply:
How so? The params are usually the most useful aspect when defining types. Try removing all the return types in this PR and you'll see that TS will be able to derive them all from the code itself rather than some override that may or may not actually widen the specificity.

Reply:
one way to think about explicit vs implicit types as just with regular variables. there's no need to write `isThing: boolean = true` or `key: string = 'foo'`, TS understands what types they are implicitly and it's the same concept here.

Reply:
Huh yeah ok. I didn't realize that TS was smart enough to infer returned types without having them be explicit. I'll refactor this.

Reply:
@ilias-t Thanks for the feedback, I've gone through and removed most of the types that I could that TS wasn't able to automatically infer.